kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -451,13 +452,19 @@ private long doCommit(
                     dataAndMetadataActions.map(
                         action -> {
                           transactionMetrics.totalActionsCounter.increment();
-                          // TODO: handle RemoveFiles.
                           if (!action.isNullAt(ADD_FILE_ORDINAL)) {
                             transactionMetrics.addFilesCounter.increment();
                             transactionMetrics.addFilesSizeInBytesCounter.increment(
                                 new AddFile(action.getStruct(ADD_FILE_ORDINAL)).getSize());
                           } else if (!action.isNullAt(REMOVE_FILE_ORDINAL)) {
                             transactionMetrics.removeFilesCounter.increment();
+                            // TODO add removeFileSizeInBytes and increment
+                            // TODO update fileSizeHistogram
+                            if (isAppendOnlyTable
+                                && new RemoveFile(action.getStruct(REMOVE_FILE_ORDINAL))
+                                    .getDataChange()) {
```

Comment:
if assigning `isRemoveDataChange = new RemoveFile(action.getStruct(REMOVE_FILE_ORDINAL)).getDataChange()`

helps make this code cleaner, I'd support it :)

if it doesn't ==> all good

Reply:
Created `RemoveFile removeFile` instead as this will be useful for the stats incrementing in follow-up PR as well